apps/dashboard/src/app/nebula-app/providers.tsx

```diff
@@ -14,17 +16,21 @@ export function NebulaProviders(props: { children: React.ReactNode }) {
   return (
     <QueryClientProvider client={queryClient}>
       <ThirdwebProvider>
-        <ThemeProvider
-          attribute="class"
-          disableTransitionOnChange
-          enableSystem={false}
-          defaultTheme="light"
-        >
-          <ToasterSetup />
-          <SanctionedAddressesChecker>
-            {props.children}
-          </SanctionedAddressesChecker>
-        </ThemeProvider>
+        <MiniAppProvider>
+          <MinikitProvider>
+            <ThemeProvider
```

Comment:
Have a separate layout for MiniApp and worldcoin to avoid including unnecessary JS in bundle for the standalone app

Reply:
You can add 2 new top level folders `farcaster-nebula-app` and `worldcoin-nebula-app` and rewrite the request to this path in middleware. 

You reuse all the components in `nebula-app` or add farcaster or worldcoin specifc tweaks too in the pages/components with this approach